### ISSUE TITLE
[console wallet] Contacts liveness does not work anymore

### ISSUE THREAD

**Issue by hansieodendaal** [tari-project/tari#5015]
The status stays on `NeverSeen`  when the contact is online

**wallet 01** to **wallet 02**
no logs of liveness events after the contacts were added

**wallet 02** to **wallet 01**
```
2022-12-07 16:16:47.219492200 [wallet:contacts_service] DEBUG New contact liveness round sent to 1 peer(s)
2022-12-07 16:16:47.219699600 [wallet:contacts_service] TRACE Liveness event 'NoMessage' for contact 🍋🚲👕🐘💤👘🎮👠🎢🍌🏆😍🐊🌵🍍🗽🍐🐚🎩👣🎷🐑🔥👹🍍😎🐬🔩🏭👛🐗🍒👢 (68bef9d7bd89c6d93716c1fb3c)  - contact was never seen
```

**Comment by SWvheerden** [comment]
I am struggling to very this, my two wallets seem fine, both connected to each other in the contact service

![Image](https://user-images.githubusercontent.com/2579813/210790153-de64ffc8-2f19-4ea3-ac37-7d64d26a6330.png)

**Comment by hansieodendaal** [comment]
Try running the wallets on different computers? I suspect that is where my issue lay.

**Comment by hansieodendaal** [comment]
After a test between @SWvheerden and myself, we found this behaviour, which may be the root cause. The onion address is differently reported by `SW wallet`, `pluto base node` and `pluto wallet`, so a connection is not possible.

SW wallet (actual):
```
│┌Connection Details───────────────────────────────────────────────────────────────────────────────────────────────────────────────┐│
││Tari Address:     8e2f659903827f894a54bb9d48280ce129fc3522779bc8d0b67c3dc4f1e66e6251                                             ││
││Node ID:        603bbedd37743d42350ee19fd7                                                                                       ││
││Network Address: /onion3/ahleygigu4tzdttchvzcg25m4bj7eop4dkzuop6t2icv7hcwmrjoxuyd:18141                                          ││
││Emoji ID:       🐺🍫🏉👕🌊🐬🐨🐴🎤🎱💦👙🎢🍞🌴😂🍟🚲🍴🍒🐙👗📌🔋💕🐞🎀💻🚒😎🐌🏁🎭                                               ││
│└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```
As perceived by pluto base node:
```
>> get-peer 8e2f659903827f894a54bb9d48280ce129fc3522779bc8d0b67c3dc4f1e66e62
Emoji ID: 🐺🍫🏉👕🌊🐬🐨🐴🎤🎱💦👙🎢🍞🌴😂🍟🚲🍴🍒🐙👗📌🔋💕🐞🎀💻🚒😎🐌🏁🐙
Public Key: 8e2f659903827f894a54bb9d48280ce129fc3522779bc8d0b67c3dc4f1e66e62
NodeId: 603bbedd37743d42350ee19fd7
Addresses:
- /onion3/l3tksj7elwt3omw3uv3gqh5gtq33zfnxviqwhckcmoquioj76j63a7ad:18141
User agent: tari/wallet/0.43.2
Features: NONE | COMMUNICATION_CLIENT
Flags: NONE
Supported protocols:
- t/msg/0.1
Last seen: 2023-01-06 14:11:10.855538300
Last updated: 2023-01-06 07:44:42 UTC (UTC)
```
As perceived by pluto wallet:
```
2023-01-09 10:45:09.002617100 [comms::connection_manager::dialer] DEBUG [Attempt 1] Connecting to peer '603bbedd37743d42'
2023-01-09 10:45:09.002654700 [comms::connection_manager::dialer] DEBUG Attempting address '/onion3/wnk2lypywo6odutjzpaouvvwsz4ux47xy6pogc334yica3zq7gqkf5yd:18141' for peer '603bbedd37743d42'
2023-01-09 10:45:10.005297200 [comms::connectivity::manager] TRACE Request: GetPeerStats(NodeId(603bbedd37743d42350ee19fd7), Sender { inner: Some(Inner { state: State { is_complete: false, is_closed: false, is_rx_task_set: true, is_tx_task_set: false } }) })
2023-01-09 10:45:12.334249400 [comms::connectivity::manager] TRACE Request: GetPeerStats(NodeId(603bbedd37743d42350ee19fd7), Sender { inner: Some(Inner { state: State { is_complete: false, is_closed: false, is_rx_task_set: true, is_tx_task_set: false } }) })
2023-01-09 10:45:23.755168800 [comms::connection_manager::dialer] DEBUG (Attempt 1) Dial failed on address '/onion3/wnk2lypywo6odutjzpaouvvwsz4ux47xy6pogc334yica3zq7gqkf5yd:18141' for peer '603bbedd37743d42' because 'Transport error for /onion3/wnk2lypywo6odutjzpaouvvwsz4ux47xy6pogc334yica3zq7gqkf5yd:18141: Host unreachable'
2023-01-09 10:45:23.755218400 [comms::connection_manager::dialer] DEBUG Failed to dial peer '603bbedd37743d42350ee19fd7' because 'ConnectFailedMaximumAttemptsReached'
2023-01-09 10:45:23.755245500 [comms::protocol::messaging::outbound] DEBUG MessagingProtocol failed to dial peer '603bbedd37743d42350ee19fd7' because 'ConnectionFailed(ConnectFailedMaximumAttemptsReached)'
```
Summary:
```
SW wallet:       /onion3/ahleygigu4tzdttchvzcg25m4bj7eop4dkzuop6t2icv7hcwmrjoxuyd:18141
pluto base node: /onion3/l3tksj7elwt3omw3uv3gqh5gtq33zfnxviqwhckcmoquioj76j63a7ad:18141
pluto wallet:    /onion3/wnk2lypywo6odutjzpaouvvwsz4ux47xy6pogc334yica3zq7gqkf5yd:18141
```

**Comment by hansieodendaal** [comment]
Upon further analysis, we found the wallet onion address changes after a restart.

**Comment by SWvheerden** [comment]
Very likely caused by: https://github.com/tari-project/tari/issues/5089

**Comment by SWvheerden** [comment]
Fixed in #5089
